### PR TITLE
Reponse with 404 if no task data found. Add cancel task

### PR DESCRIPTION
**Pull request by vkocheryzhkin**

### CODE REVIEW

File: project/main.py [hunk 1]
```diff
@@ -54,11 +56,22 @@ def get_task_status(id):
...
 
 @app.get("/task/{id}/statistics-results")
 def get_stats_result(id):
+    if not task_exists(id):
+        return JSONResponse(status_code=404)
     return StreamingResponse(iterfile(f'./tmp/{id}/statistics_output.json'), media_type="application/octet-stream")
 
+@app.get("/task/{id}/cancel")
+def cancel_task(id):
+    if not task_exists(id):
```
Comment (adavydenko) [inline project/main.py]: should validate by celery whether task exists without file structure
its theoretically possible the directory does not exist, but celery runs task in background, hence we shall stop that task

File: project/main.py [hunk 2]
```diff
@@ -54,11 +56,22 @@ def get_task_status(id):
 
 @app.get("/task/{id}/result")
 def get_schedule_result(id):
-    # return StreamingResponse(iterfile(f'{id}_rostering.json'), media_type = "application/json")
+    if not task_exists(id):
```
Comment (adavydenko) [inline project/main.py]: if not task_exists And result available

File: project/main.py [hunk 3]
```diff
@@ -54,11 +56,22 @@ def get_task_status(id):
...
+        return JSONResponse(status_code=404)
+
     return StreamingResponse(iterfile(f'./tmp/{id}/rostering.json'), media_type = "application/octet-stream")
 
 
 @app.get("/task/{id}/statistics-results")
 def get_stats_result(id):
+    if not task_exists(id):
```
Comment (adavydenko) [inline project/main.py]: + and stats file available

**Review by adavydenko — COMMENTED** [review]
please check comments

File: project/main.py (hunk 1 above)
Comment (vkocheryzhkin) [reply project/main.py]: celery task owns dir creation on start here https://github.com/gitpay-ru/workforce-mvp/blob/master/project/worker.py#L102

File: project/main.py (hunk 2 above)
Comment (vkocheryzhkin) [reply project/main.py]: task_exists just checks that dir exists. no dir == no results

File: project/main.py (hunk 3 above)
Comment (vkocheryzhkin) [reply project/main.py]: no dir - no results

File: project/main.py [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
 app.mount("/static", StaticFiles(directory="static"), name="static")
 templates = Jinja2Templates(directory="templates")
 
-# @app.get("/")
-# def home(request: Request):
-#     return templates.TemplateResponse("home.html", context={"request": request})
-
+def task_exists(id):
+    t = Path(f'./tmp/{id}')
+    return t.exists()
```
Comment (vkocheryzhkin) [inline project/main.py]: @adavydenko here it just checks for folder

File: project/main.py (hunk 1 above)
Comment (adavydenko) [reply project/main.py]: *if* there is a dedicated straigthforward api to check task status - its better to use that without 3rd statuses, 
consider using https://docs.celeryq.dev/en/stable/userguide/tasks.html#built-in-states if applicable

File: project/main.py (hunk 2 above)
Comment (adavydenko) [reply project/main.py]: if dir exists - this doesn't mean, the results are available. should handle that gently

File: project/main.py (hunk 3 above)
Comment (adavydenko) [reply project/main.py]: dir could exist without result

**Review by adavydenko — APPROVED** [review]
Looks good!

**Pull request merged**